### ISSUE TITLE
Makefile improvements

### ISSUE THREAD

**Issue by freeflyclone** [freeflyclone/madstyletv#1]
The Makefile set for linux builds is sub-optimal.  There is a lot of boilerplate amongst the various examples and the xclass libraries builds.  A set of include.mk files would go a long way towards eliminating the cruft.

**Comment by freeflyclone** [comment]
Added a couple of .mk includes for the Makefile build system under linux.  Functionality remains the same as before, but with way less duplication of makefile cruft.

**Comment by freeflyclone** [comment]
Closing this.  The makefile builds are sufficient at this point.

**Comment by freeflyclone** [comment]
Heh.  Learning GitHub's issue tracker.
Anyway... makefile refactoring got closed prematurely.  I've refactored it significantly to achieve repeatability and to ease creation of new example programs.  Specificaly, the Makefile for an individual example now only lists the source files and additional libraries that are needed.  All the recipies are contained in project wide makefile includes at strategic points in the source tree.  At this point, one can type "make clean ; make" at any working point in the source tree and everything from that point down will get rebuild.  Conversely, just "make" will do an incremental build from that point down.  Like make is supposed to work.
